stl/inc/flat_map

```diff
@@ -0,0 +1,873 @@
+    flat_map(initializer_list<value_type> _I, const key_compare& _Comp, const Allocator& _Alloc)
+        : flat_map(_I.begin(), _I.end(), _Comp, _Alloc) {}
+
+    template <_Valid_Allocator_for_flat_map<key_container_type, mapped_container_type> Allocator>
+    flat_map(initializer_list<value_type> _I, const Allocator& _Alloc) : flat_map(_I, key_compare(), _Alloc) {}
+
+    flat_map(sorted_unique_t _S, initializer_list<value_type> _I, const key_compare& _Comp = key_compare())
+        : flat_map(_S, _I.begin(), _I.end(), _Comp) {}
+
+    template <_Valid_Allocator_for_flat_map<key_container_type, mapped_container_type> Allocator>
+    flat_map(sorted_unique_t _S, initializer_list<value_type> _I, const key_compare& _Comp, const Allocator& _Alloc)
+        : flat_map(_S, _I.begin(), _I.end(), _Comp, _Alloc) {}
+
+    template <_Valid_Allocator_for_flat_map<key_container_type, mapped_container_type> Allocator>
+    flat_map(sorted_unique_t _S, initializer_list<value_type> _I, const Allocator& _Alloc)
+        : flat_map(_S, _I, key_compare(), _Alloc) {}
+
+    flat_map& operator=(initializer_list<value_type> _I) {
+        clear();
+        insert(_I.begin(), _I.end());
+        return *this;
+    }
+
+    // [container.reqmts] iterators
+    _NODISCARD iterator begin() noexcept {
+        return iterator{_Data.keys.cbegin(), _Data.values.begin()};
+    }
+
+    _NODISCARD const_iterator begin() const noexcept {
+        return const_iterator{_Data.keys.cbegin(), _Data.values.begin()};
+    }
+
+    _NODISCARD iterator end() noexcept {
+        return iterator{_Data.keys.cend(), _Data.values.end()};
+    }
+
+    _NODISCARD const_iterator end() const noexcept {
+        return const_iterator{_Data.keys.cend(), _Data.values.end()};
+    }
+
+    _NODISCARD reverse_iterator rbegin() noexcept {
+        return _STD make_reverse_iterator(end());
+    }
+
+    _NODISCARD const_reverse_iterator rbegin() const noexcept {
+        return _STD make_reverse_iterator(end());
+    }
+
+    _NODISCARD reverse_iterator rend() noexcept {
+        return _STD make_reverse_iterator(begin());
+    }
+
+    _NODISCARD const_reverse_iterator rend() const noexcept {
+        return _STD make_reverse_iterator(begin());
+    }
+
+    _NODISCARD const_iterator cbegin() const noexcept {
+        return const_iterator{_Data.keys.cbegin(), _Data.values.cbegin()};
+    }
+
+    _NODISCARD const_iterator cend() const noexcept {
+        return const_iterator{_Data.keys.cend(), _Data.values.cend()};
+    }
+
+    _NODISCARD const_reverse_iterator crbegin() const noexcept {
+        return _STD make_reverse_iterator(cend());
+    }
+
+    _NODISCARD const_reverse_iterator crend() const noexcept {
+        return _STD make_reverse_iterator(cbegin());
+    }
+
+    void swap(flat_map& _Other) noexcept {
+        _STD swap(_Key_compare, _Other._Key_compare);
+        _Data.keys.swap(_Other._Data.keys);
+        _Data.values.swap(_Other._Data.values);

```

Comment:
Is there any difference between `std::swap` and `std::ranges::swap` for non-range types? Or could `_Key_compare` be a range?

Reply:
> Is there any difference between `std::swap` and `std::ranges::swap` for non-range types? Or could `_Key_compare` be a range?

`ranges::swap` doesn't do anything special for range types. It may choose [an ADL-found `swap` overload](https://eel.is/c++draft/concept.swappable#2.1) for `_Key_compare`.